# Cars Dataset
Here, 
The data of different cars is given with their specifications.

In [1]:
import pandas as pd
car= pd.read_csv("Project+2+-+Cars+Dataset.csv")

In [2]:
car

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        21           28   
424  $40,083          2.3        5.0         242        20           26   
425  $42,573          2.9        6.0         268        19           26   
426  $24,641          1.9        4.0         170        22           29   
427  $33,112          2.5        5.0         208        20           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[428 rows x 15 columns]

In [3]:
car.head(2)

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   

   Wheelbase  Length  
0        106     189  
1        101     172

In [17]:
car.shape

(432, 15)

# 1) Instruction ( For Data Cleaning ) 
- Find all Null Values in the dataset. If there is any null value in any column, then fill it with the mean of that column.

In [4]:
car.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      2
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

In [6]:
car["Cylinders"].fillna(car["Cylinders"].mean())

0      6.0
1      4.0
2      4.0
3      6.0
4      6.0
      ... 
423    5.0
424    5.0
425    6.0
426    4.0
427    5.0
Name: Cylinders, Length: 428, dtype: float64

In [7]:
car.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      0
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

In [8]:
car["Cylinders"].fillna(car["Cylinders"].mean(), inplace = True)

In [9]:
car.isnull().sum()

Make           0
Model          0
Type           0
Origin         0
DriveTrain     0
MSRP           0
Invoice        0
EngineSize     0
Cylinders      0
Horsepower     0
MPG_City       0
MPG_Highway    0
Weight         0
Wheelbase      0
Length         0
dtype: int64

# Q2.) Based on value counts
  Check what are the different types of Make are there in  dataset. And,, what is the count(occurence) of each make in the data.

In [10]:
car.head(2)

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   

   Wheelbase  Length  
0        106     189  
1        101     172

In [13]:
car["Make"].value_counts()

Toyota           28
Chevrolet        27
Mercedes-Benz    26
Ford             23
BMW              20
Audi             19
Honda            17
Nissan           17
Volkswagen       15
Chrysler         15
Dodge            13
Mitsubishi       13
Volvo            12
Jaguar           12
Hyundai          12
Subaru           11
Pontiac          11
Mazda            11
Lexus            11
Kia              11
Buick             9
Mercury           9
Lincoln           9
Saturn            8
Cadillac          8
Suzuki            8
Infiniti          8
GMC               8
Acura             7
Porsche           7
Saab              7
Land Rover        3
Oldsmobile        3
Jeep              3
Scion             2
Isuzu             2
MINI              2
Hummer            1
Name: Make, dtype: int64

# 3) Instruction ( Filtering )
- Show all the records where Origin is Asia.

In [14]:
car.head(2)

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   

   Wheelbase  Length  
0        106     189  
1        101     172

In [18]:
car[car["Origin"].isin(['Asia'])]

Make                       Model    Type Origin DriveTrain      MSRP  \
0     Acura                         MDX     SUV   Asia        All  $36,945    
1     Acura              RSX Type S 2dr   Sedan   Asia      Front  $23,820    
2     Acura                     TSX 4dr   Sedan   Asia      Front  $26,990    
3     Acura                      TL 4dr   Sedan   Asia      Front  $33,195    
4     Acura                  3.5 RL 4dr   Sedan   Asia      Front  $43,755    
..      ...                         ...     ...    ...        ...       ...   
396  Toyota  MR2 Spyder convertible 2dr  Sports   Asia       Rear  $25,130    
397  Toyota                      Tacoma   Truck   Asia       Rear  $12,800    
398  Toyota       Tundra Regular Cab V6   Truck   Asia       Rear  $16,495    
399  Toyota    Tundra Access Cab V6 SR5   Truck   Asia        All  $25,935    
400  Toyota                   Matrix XR   Wagon   Asia      Front  $16,695    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        17           23   
1    $21,761          2.0        4.0         200        24           31   
2    $24,647          2.4        4.0         200        22           29   
3    $30,299          3.2        6.0         270        20           28   
4    $39,014          3.5        6.0         225        18           24   
..        ...         ...        ...         ...       ...          ...   
396  $22,787          1.8        4.0         138        26           32   
397  $11,879          2.4        4.0         142        22           27   
398  $14,978          3.4        6.0         190        16           18   
399  $23,520          3.4        6.0         190        14           17   
400  $15,156          1.8        4.0         130        29           36   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
396    2195         97     153  
397    2750        103     191  
398    3925        128     218  
399    4435        128     218  
400    2679        102     171  

[158 rows x 15 columns]

158  

# 4) Instruction ( Removing unwanted records )
- Remove all the records (rows) where Weight is above 4000.

In [20]:
car.head(2)

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   

   Wheelbase  Length  
0        106     189  
1        101     172

In [28]:
# car[~(car["Weight"]>4000)]

In [27]:
after removing the weight above 4000, the left out cars are 
428-103

325

# 5) Instruction ( Applying function on a column )
- Increase all the values of 'MPG_City' column by 3.

In [30]:
car.head(2)

Make           Model   Type Origin DriveTrain      MSRP   Invoice  \
0  Acura             MDX    SUV   Asia        All  $36,945   $33,337    
1  Acura  RSX Type S 2dr  Sedan   Asia      Front  $23,820   $21,761    

   EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  Weight  \
0         3.5        6.0         265        17           23    4451   
1         2.0        4.0         200        24           31    2778   

   Wheelbase  Length  
0        106     189  
1        101     172

In [31]:
car["MPG_City"] = car["MPG_City"].apply(lambda x:x+3)

In [32]:
car

Make                    Model   Type  Origin DriveTrain      MSRP  \
0    Acura                      MDX    SUV    Asia        All  $36,945    
1    Acura           RSX Type S 2dr  Sedan    Asia      Front  $23,820    
2    Acura                  TSX 4dr  Sedan    Asia      Front  $26,990    
3    Acura                   TL 4dr  Sedan    Asia      Front  $33,195    
4    Acura               3.5 RL 4dr  Sedan    Asia      Front  $43,755    
..     ...                      ...    ...     ...        ...       ...   
423  Volvo  C70 LPT convertible 2dr  Sedan  Europe      Front  $40,565    
424  Volvo  C70 HPT convertible 2dr  Sedan  Europe      Front  $42,565    
425  Volvo               S80 T6 4dr  Sedan  Europe      Front  $45,210    
426  Volvo                      V40  Wagon  Europe      Front  $26,135    
427  Volvo                     XC70  Wagon  Europe        All  $35,145    

      Invoice  EngineSize  Cylinders  Horsepower  MPG_City  MPG_Highway  \
0    $33,337          3.5        6.0         265        20           23   
1    $21,761          2.0        4.0         200        27           31   
2    $24,647          2.4        4.0         200        25           29   
3    $30,299          3.2        6.0         270        23           28   
4    $39,014          3.5        6.0         225        21           24   
..        ...         ...        ...         ...       ...          ...   
423  $38,203          2.4        5.0         197        24           28   
424  $40,083          2.3        5.0         242        23           26   
425  $42,573          2.9        6.0         268        22           26   
426  $24,641          1.9        4.0         170        25           29   
427  $33,112          2.5        5.0         208        23           27   

     Weight  Wheelbase  Length  
0      4451        106     189  
1      2778        101     172  
2      3230        105     183  
3      3575        108     186  
4      3880        115     197  
..      ...        ...     ...  
423    3450        105     186  
424    3450        105     186  
425    3653        110     190  
426    2822        101     180  
427    3823        109     186  

[428 rows x 15 columns]

# By Vibhor Nanda